pkg/prometheus/agent/daemonset.go

```diff
@@ -55,7 +54,7 @@ func makeDaemonSet(
 
 	operator.UpdateObject(
 		daemonSet,
-		operator.WithName(name),
+		operator.WithName("agent-daemonset"),
```

Comment:
Could you clarify what is the goal here?

If a daemonset has a hardcoded name, it will be impossible to deploy 2 daemonsets in the same namespace 🤔

Reply:
Maybe `agent-daemonset-$NODE_NAME`? We already can retrieve `NODE_NAME` from config-reloader: https://github.com/prometheus-operator/prometheus-operator/pull/6636

Reply:
Why not re-use the agent name from the PrometheusAgent resource, like we do with statefulset?

Reply:
agreed with Arthur, we need a non-unique name. It should be possible to deploy 2 daemonset agents on the same cluster (though it's a niche scenario). IIUC it's also inconsistent with the logic of keyToDaemonSetKey(), meaning that the operator can't link back to the DaemonSet object from a PrometheusAgent name.